docs/talkback-guidelines.md

```diff
@@ -0,0 +1,401 @@
+
+#### Governing Accessibility Principles 
+
+* Make the information on the screen as visible as possible. 
+* Always design the size of controls and touch areas inclusive of all use cases.
+* Provide as much support context and details for all views & actions. Images, buttons, lists and the actions they enable should have thorough, well-formatted descriptions that allow the accessibility APIs to deliver their intent seamlessly. 
+
+	For more details visit [Make apps more accessible
+](https://developer.android.com/guide/topics/ui/accessibility/apps.html).
+
+#### TalkBack
+This document will provide all the resources needed to improve an app's accessibility using TalkBack. If you have never worked with TalkBack you can visit the "Using TalkBack" document to get started.
+
+- [Using TalkBack](using-talback.md)
+
+## <a name="guidelines"></a>Guidelines
+
+### <a name="basics"></a>Basics
+
+The accessibility framework within Android has several ways in which you can provide cues to TalkBack, so it knows how it makes it's announcement when a view becomes focused. 
+
+* [`contentDescription`](https://developer.android.com/reference/android/view/View#attr_android:contentDescription)  - Each view has this attribute that allows you to set meaningful and descriptive labels. If this is missing TalkBack might not be able to provide much value to the user. 
+* [`importantForAccessibility`](https://developer.android.com/reference/android/view/View#attr_android:importantForAccessibility) - In cases where a view or layout might not need to be accessible, this attribute can be set. It allows TalkBack to ignore views that have this property set to `yes`, thus eliminating the need for a content description. 
+* [`hint`](https://developer.android.com/reference/android/widget/TextView#attr_android:hint) - For usability purposes, the text input control `TextView` has the hint attribute. If another view is providing the label, then you can utilize the [`labelFor`](http://developer.android.com/reference/android/view/View.html#setLabelFor(int)) attribute. Below is an example.  
+
+ ```xml
+ <LinearLayout
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+    <TextView
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/email_subject_label"
+        android:labelFor="@id/email_subject" />
+    <EditText
+        android:id="@+id/email_subject"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content" />
+</LinearLayout>
+```
+
+#### Labels / Content Description 
+To make a view ready for accessibility these are the rules that should govern the content descriptions that are set. 
+
+* Do not include the type of the control. TalkBack will automatically announce the description you have set and the type. 
```

Comment:
By `Do not include the type of the control.` you mean Not to include the type of control into the accessibility announcement? Or something else?

Reply:
Yes, that's what I meant since `TalkBack` determines that itself and if needed that can be customized using by modifying the `AccessibilityNodeInfo`